Widget/app/src/main/java/com/microsoft/device/display/samples/widget/feed/RssFeed.kt

```diff
@@ -149,40 +151,40 @@ object RssFeed {
         }
     }
 
-    private fun shouldInsertItemInRssItems(tag: String, rssItem: RssItem?): Boolean {
+    fun shouldInsertItemInRssItems(tag: String, rssItem: RssItem?): Boolean {
         return tag.equals(ITEM, ignoreCase = true) && rssItem != null
     }
 
-    private fun isEndOfFeed(tag: String): Boolean {
+    fun isEndOfFeed(tag: String): Boolean {
         return tag.equals(CHANNEL, ignoreCase = true)
     }
 
-    private fun parseTitle(title: String): String {
+    fun parseTitle(title: String): String {
         return stripHtml(title)
     }
 
-    private fun parseDescription(description: String): String {
+    fun parseDescription(description: String): String {
         val descriptionWithEnding = description.substring(
             0,
-            Math.min(description.length, DESCRIPTION_MAX_CHARS)
+            min(description.length, DESCRIPTION_MAX_CHARS)
         ) + "..."
         return stripHtml(descriptionWithEnding)
     }
 
-    private fun parseDate(date: String): String {
+    fun parseDate(date: String): String {
         return date.substring(0, date.length - 6)
     }
 
-    private fun stripHtml(html: String): String {
+    fun stripHtml(html: String): String {
         return Html.fromHtml(html, Html.FROM_HTML_MODE_COMPACT)
             .toString()
             .replace(
-                "&.*?;".toRegex(),
+                "[&.*?;]".toRegex(),
                 ""
             )
     }
 
-    private fun getFeedUrlFromPreferences(context: Context): String {
```

Comment:
Do we need that all functions become `public`? we should be able to test the parser as a black box that given an input and its output we test that what we get is correct. All internals about how the parser work (functions) should be hidden to us and don't really matter. 
Agree though that current design needs changes to allow configuration and injection of the (mock) network engine. Or we even can decouple the parser engine from the network engine, we probably should do that since both should be independent from each other in its core functionality.

Reply:
Great point you made, I refactored all internal methods to have them private again and only used black-box testing. Also separated the two functionalities - network and parsing to have them decoupled. Thank you for the suggestions, really helpful! 😄